### PR TITLE
leftwm-theme: init at 0.1.2, leftwm-config: init at 0-unstable-2024-03-23

### PR DESCRIPTION
**Pull request by Denperidge**
## Description of changes
Implements leftwm-theme and leftwm-config. Based on #211955 , re-do of #317279 !
Due to the original PR being about a year old, I've temporarily removed him from the maintainers meta for lightwm-theme & config. If he still has an interest in that'd be great! And if he doesn't, thanks for putting the original PR up!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Sigmanificient** [comment]
I would just have kept the commit history as follows:
- `maintainers: add denperidge`
- `leftwm-theme: init at 0.1.2`
- `leftwm-config: init at 0-unstable-2024-03-23`

or if you really want to keeps yanganto's commits:
- (`leftwm-config: init at 0.1.0`)
- (`leftwm-theme: init at unstable-2022-12-24`)
- `maintainers: add denperidge`
- `leftwm-{config,theme}: update, migrate to by-name, refactor`

**Comment by Sigmanificient** [comment]
Not sure why force pushing would have autoclosed #317279 as it's a pretty common thing in nixpkgs prs
May we keep things in the previous pr to keep previous records?

**Comment by Denperidge** [comment]
> Not sure why force pushing would have autoclosed #317279 as it's a pretty common thing in nixpkgs prs May we keep things in the previous pr to keep previous records?

Sure thing! I'll try to push to there. Do I remove the comment in the git commit as well?

**Comment by Sigmanificient** [comment]
> > Not sure why force pushing would have autoclosed #317279 as it's a pretty common thing in nixpkgs prs May we keep things in the previous pr to keep previous records?
> 
> Sure thing! I'll try to push to there. Do I remove the comment in the git commit as well?

Comments are fine ig

**Comment by Denperidge** [comment]
> Not sure why force pushing would have autoclosed #317279 as it's a pretty common thing in nixpkgs prs May we keep things in the previous pr to keep previous records?

Locked out of re-opening the PR after force pushing. Do I try force pushing to here? My apologies for the documentation soup!
image

**Comment by Sigmanificient** [comment]
Hmm lets keep that one them

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-config/package.nix:3]: I believe this `rec` is unused anymore

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Comment by Sigmanificient** [comment]
Result of `nixpkgs-review pr 317805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>leftwm-theme</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>leftwm-config</li>
  </ul>
</details>

File: pkgs/by-name/le/leftwm-theme/package.nix
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-theme/package.nix:10]: Missed the `v` in rev (my bad on previous suggestion)

```suggestion
    rev = "refs/tags/v${version}";
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:3]: On it!

File: pkgs/by-name/le/leftwm-theme/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-theme/package.nix:10]: Aand also on it!

**Comment by Denperidge** [comment]
Made the changes! If want/need be, I'll try to do a force push with the last commit merged with the rest, otherwise it should hopefully be ok to merge!

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-config/package.nix:17]: I think the parenthesis are uneeded

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:17]: Oh yeah, is a remainder from originally. I think it kinda works for readability's sake, but I can remove it if preferred 😌 @Sigmanificient

Comment (Sigmanificient) [reply pkgs/by-name/le/leftwm-config/package.nix:17]: Prefer to wait for someone with more experience to give yu a better answer

**Comment by Denperidge** [comment]
As recommended on the nixpkgs dev matrix, broken has been temporarily removed due to a lack of current findings on aarch64 being broken. If ofborg clears it, then the parenthesis question is no longer relevant

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:17]: Resolving conversation due to below!

**Comment by Denperidge** [comment]
Ofborg cleared it! Should be ready for merge

File: pkgs/by-name/le/leftwm-theme/package.nix
Comment (OPNA2608) [inline pkgs/by-name/le/leftwm-theme/package.nix:11]: You didn't update the hash here when you bumped the version.

```
error: hash mismatch in fixed-output derivation '/nix/store/5q2c69aqk8jl0830588sqr07004l80ss-source.drv':
         specified: sha256-tYT1eT7Rbs/6zZcl9eWsOA651dUGoXc7eRtVK8fn610=
            got:    sha256-JCNpgdV4v0DzRmELAv+UwPAZ7O2V0KI4t8zirhhQshA=
error: 1 dependencies of derivation '/nix/store/86c92zvq6cnsw6byvr3m6fzknnaqkcbl-leftwm-theme-0.1.2-vendor.tar.gz.drv' failed to build
```

Comment (Denperidge) [reply pkgs/by-name/le/leftwm-theme/package.nix:11]: Wow, I was not able to replicate this using `nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}' --check` until I specifically removed the previous hash, gave no error before. Fixing now!

**Comment by Denperidge** [comment]
While fixing the hash, noticed leftwm-theme had a vulnerability update that didn't get included in 0.1.2, so updated to that one in one go. The date on the first commit message was wrong seemingly, so I squashed and reworded them like @Sigmanificient suggested in my previous PR

File: pkgs/by-name/le/leftwm-theme/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/le/leftwm-theme/package.nix:31]:
```suggestion
    description = "Theme manager for LeftWM";
```

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/le/leftwm-config/package.nix:18]:
```suggestion
    description = "Little satellite utility for LeftWM";
```

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/le/leftwm-config/package.nix:17]: We should add a really short comment why those failed

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/le/leftwm-config/package.nix:14]:
```suggestion
  cargoHash = "sha256-U3mgbG9h2cDqr0aqxbI2CJUOweIoDXDxmsWg0zxolSo=";
```

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:17]: I think that's a leftover from the original commits i based the pr on, but I'm gonna run it on an aarch64 to confirm and/or get output logs

File: pkgs/by-name/le/leftwm-theme/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-theme/package.nix:31]: On it!

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:14]: On it!

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:18]: Aand on it!

File: pkgs/by-name/le/leftwm-config/package.nix
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:17]: Update: I checked above, I have definitely tested this before on leftwm-theme and removed it, but didn't see it was on config too. Gonna build to be sure, but it might be okay for removal

Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:17]: Aaand it works! Removed

**Review by Sigmanificient — APPROVED** [review]

**Pull request merged**
